File: testcases/x86_testcases/x86_output/test7/test7.c
#include <stdio.h>
#include <stdlib.h>

// Define the MathOperation class
typedef struct {
    // No need to declare instance variables for this example
} MathOperation;

// Define the constructor for MathOperation
void MathOperation___init__(MathOperation* obj) {
    printf("Inside the constructor for MathOperation class\n");
    return;
}

// Define the uncalled method for MathOperation
void MathOperation_uncalled(MathOperation* obj, int x) {
    printf("This is never called\n");
    return;
}

// Define the main function
int main() {
    printf("This is a simple testcase for classes\n");

    // Creating an instance of MathOperation
    MathOperation math;
    MathOperation___init__(&math);

    return 0;
}

File: testcases/x86_testcases/test7.py
class MathOperation:
    def __init__(self) -> None:
        self.lifespan : int = 0
        print("Inside the constructor for MathOperation class")
        return

    def uncalled(self, x : int) -> None:
        print("This is never called")
        return

def main():
    print("This is a simple testcase for classes")
    math : MathOperation = MathOperation()
    return

if __name__ == "__main__":
    main()
